Keas.Mvc/Controllers/TeamController.cs

```diff
@@ -105,5 +110,141 @@ private bool TeamExists(int id)
             return _context.Teams.Any(e => e.Id == id);
         }
+
+        public IActionResult CreateGroup()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CreateGroup(Group model)
+        {
+            if (await _context.Groups.AnyAsync(a => a.Name.Equals(model.Name, StringComparison.OrdinalIgnoreCase)))
+            {
+                ModelState.AddModelError("Name", "Group name already used");
+            }
+
+            if (ModelState.IsValid)
+            {
+                _context.Add(model);
+                await _context.SaveChangesAsync();
+                return RedirectToAction("Index", "Admin");
+            }
+            return View(model);
+        }
+
+        public async Task<IActionResult> GroupDetails(int id)
+        {
+            var group = await _context.Groups.Include(a => a.GroupPermissions).ThenInclude(a => a.User).Include(a => a.Teams).ThenInclude(a => a.Team).AsNoTracking().SingleAsync(a => a.Id == id);
+
+            return View(group);
+        }
+
+        public async Task<IActionResult> AddGroupUser(int id)
+        {
+            var model = new GroupUserPostModel();
+            model.Group = await _context.Groups.AsNoTracking().SingleAsync(a => a.Id == id);
+            return View(model);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> AddGroupUser(GroupUserPostModel model)
+        {
+            var group = await _context.Groups.AsNoTracking().SingleAsync(a => a.Id == model.Group.Id);
+
+            var user = await _context.Users.SingleOrDefaultAsync(u => u.Email == model.UserEmail || u.Id == model.UserEmail);
+
+            if (user == null)
+            {
+                if (model.UserEmail.Contains("@"))
+                {
+                    user = await _userService.CreateUserFromEmail(model.UserEmail);
+                }
+                else
+                {
+                    user = await _userService.CreateUserFromKerberos(model.UserEmail);
+                }
+            }
+
+            if (user == null)
+            {
+                ErrorMessage = "User not found";
+                return View(model);
+            }
+
+            if (await _context.GroupPermissions.AnyAsync(a => a.GroupId == model.Group.Id && a.UserId == user.Id))
+            {
+                Message = "User Already Exists for group";
+                return RedirectToAction("GroupDetails", "Team", new {id = model.Group.Id});
+            }
+
+            
+
+            var groupPermission = new GroupPermission{ GroupId = group.Id, UserId = user.Id };
+
+            _context.GroupPermissions.Add(groupPermission);
+            await _context.SaveChangesAsync();
+            Message = "User " + user.Name + " has been added.";
+            return RedirectToAction("GroupDetails", "Team", new { id = model.Group.Id });
+        }
+
+        public async Task<IActionResult> RemoveGroupie(int id, int groupId)
```

Comment:
why remove grupie but not add groupie? or does the group user get added but then they become groupies and have to be removed?

Reply:
No one should add groupies :) Was just having fun. Can change it.